### PR TITLE
Replace express req.params 'any' type with index types (Revive #17610)

### PR DESCRIPTION
**Pull request by felipeochoa**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: #17610
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by cyrilschumacher — APPROVED** [review]

**Review by dolanmiu — APPROVED** [review]

**Comment by dolanmiu** [comment]
Good work

File: types/express/express-tests.ts [hunk 1]
```diff
@@ -142,8 +149,8 @@ import * as http from 'http';
...
 namespace node_tests {
     {
-        // http.createServer can take express application
-        const app: express.Application = express();
-        http.createServer(app).listen(5678);
-    }
+    // http.createServer can take express application
+    const app: express.Application = express();
+    http.createServer(app).listen(5678);
+}
```
Comment (erikma) [inline types/express/express-tests.ts]: Unneeded unindent in this block

**Review by erikmav — APPROVED** [review]
One small formatting issue otherwise LGTM

**Review by MichalLytek — CHANGES_REQUESTED** [review]
I'm afraid we will se this again:
> This is breaking my app's build, which uses Typescript 2.2.2

@mhegazy can we safely bump all deps to TS 2.3?

**Review by 0x6368656174 — APPROVED** [review]

**Review by kvcpr — APPROVED** [review]

**Review by pinguet62 — APPROVED** [review]

**Pull request closed without merging**